src/garage/torch/algos/vpg.py

```diff
@@ -432,7 +483,7 @@ def process_samples(self, itr, paths):
         for path in paths:
             if 'returns' not in path:
                 path['returns'] = tu.discount_cumsum(path['rewards'],
-                                                     self.discount)
+                                                     self.discount).copy()
```

Comment:
why copy this?

Reply:
If not, I got the error
`E       ValueError: some of the strides of a given numpy array are negative. This is currently not supported, but will be added in future releases.`
This is becaue we revert the order of output `[::-1]` in the `discount_cumsum` function.
So I need to copy the array

Reply:
How did it work before? What changed?

Reply:
Can you use some other numpy function to get a reversed view, other than negative indexing?

When you call `copy()` you break the gradient path. If your returns have a differentiable components, e.g. a differentiable reward augmentation, your augmented rewards will no longer be differentiable.

Reply:
This error is from [here](https://github.com/rlworkgroup/garage/blob/a2207eeff9fe7ff84c9db68202ffa9f0f233ac22/src/garage/torch/algos/vpg.py#L503).
Previously, there is only LinearFeatureBaseline which doesn't have to convert numpy.array to torch.tensor. But now we need to convert numpy.array to torch.tensor for GaussianMLPValueFunction.

Reply:
According to this [post](https://discuss.pytorch.org/t/torch-from-numpy-not-support-negative-strides/3663/11), problem is in PyTorch. PyTorch doesn't support numpy.array with negative stride.
For the gradient path, the returns values from the `tu.discount_cumsum` is numpy.array not a torch.tensor, so I think there is no gradient path for the values yet. 

Also I'm just curious that we use rewards from environment to compute the return values and there is module or functions that compute gradient for the rewards?